plumbing/transport/ssh/common.go

```diff
@@ -166,10 +168,36 @@ func dial(network, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
 	}
 	defer cancel()
 
-	conn, err := proxy.Dial(ctx, network, addr)
+	var conn net.Conn
+	var err error
+
+	if proxyOpts.URL != "" {
+		proxyUrl, err := url.Parse(proxyOpts.URL)
+		if err != nil {
+			return nil, err
+		}
+		if proxyOpts.Username != "" && proxyOpts.Password != "" {
```

Comment:
Password does not seem to be a required field for proxy auth purposes.

```suggestion
		if proxyOpts.Username != "" {
```

Reply:
is this true for ssh only or also for http?

Reply:
AFAIK both. I could not find any official documentation to back that up either way though.